### PR TITLE
Create Your Own Enumerable

### PR DESCRIPTION
**Pull request by DammyShittu**
## In this project, we completed the following requirements;

- Create a class `MyList` that has an instance variable `@list`.

- In `MyList` implement a method `#each` that yields successive members of `@list` and uses the `MyEnumerable` module.

- Create a module `MyEnumerable` that implements the following methods:
    - `#all?`
    - `#any?`
    - `#filter`
- Each class and module should have a separate `.rb` file.

- Verify your solution:

### Linters were also correctly set up.

### CODE REVIEW

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  include MyEnumerable
+
+  def initialize(*args)
+    @list = args
+  end
+
+  def each(&block)
+    @list.each(&block)
+  end
+end
```
Comment (mikethreels) [inline my_list.rb]: - good job on creating the MyList class 👍 however the `#each` method you created here should be created by yourselves not using the existing `#each` enumerable.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: my_list.rb (hunk 1 above)
Comment (rpire) [reply my_list.rb]: Hello, @mikethreels 👋

Thank you for reviewing our work. I have a question regarding the required changes that you have proposed. Would implementing a `.map` method do for creating the `#each` method in the `MyList` class?

Or maybe, should we implement it using a `for` loop and the `@list.length`?

Thanks in advance for your help! Cheers and Happy coding to you too! 🤓

Comment (mikethreels) [reply my_list.rb]: Hi @rpire 👋 

Try implementing it using a loop `for` or `while` using the `@list.length`.
looking at your other methods this should be a walk in the park for you 🥇 

You're doing great.
let me know if you have any additional questions!

Comment (rpire) [reply my_list.rb]: Greetings again, @mikethreels 👋

About my last question, we have decided to implement a `while` loop. Using the `.map` method was the same as using the `.each` method, and using a `for` loop would give Linter errors.

Required changes completed! 🙌

**Comment by rpire** [comment]
Once again, thank you for reviewing our work. 🔎

We will proceed to ask for another code review.

Cheers and Happy coding! 🤓

**Review by mikethreels — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 :
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rpire** [comment]
### Thank you 🙌

We will proceed to merge these changes into the `dev` branch. 🚀

Best regards!

**Pull request merged**